source/game/g_svcmds.cpp

```diff
@@ -259,11 +282,24 @@ void SV_WriteIPList( void )
 	{
 		if (ipfilters[i].timeout && ipfilters[i].timeout <= game.serverTime)
 			continue;
-		*(unsigned *)b = ipfilters[i].compare;
-		if( ipfilters[i].timeout )
-			Q_snprintfz( string, sizeof( string ), "addip %i.%i.%i.%i %.2f\r\n", b[0], b[1], b[2], b[3], (ipfilters[i].timeout - game.serverTime)/(1000.0f*60.0f) );
-		else
-			Q_snprintfz( string, sizeof( string ), "addip %i.%i.%i.%i\r\n", b[0], b[1], b[2], b[3] );
+
+
+		float timeout = (ipfilters[i].timeout - game.serverTime)/(1000.0f*60.0f);
+		if (ipfilters[i].steamban) {
+			uint64_t id = ipfilters[i].steamid;
+
+			const char *type = ipfilters[i].ismute ? "mute" : "ban";
+			if( ipfilters[i].timeout )
+				Q_snprintfz( string, sizeof( string ), "%s %llu %.2f\r\n", type, id, timeout);
+			else
+				Q_snprintfz( string, sizeof( string ), "%s %llu\r\n", type, id );
+		}else{
+			*(unsigned *)b = ipfilters[i].compare;
```

Comment:
I realize that you didn't write this, but it probably doesn't work as intended?
You are treating an `unsigned int` as an array of 4 `unsigned int`s.